src/PIL/ImageFont.py

```diff
@@ -215,13 +215,123 @@ def getmetrics(self):
         """
         return self.font.ascent, self.font.descent
 
+    def getlength(self, text, mode="", direction=None, features=None, language=None):
+        """
+        Returns length (in pixels) of given text if rendered in font with
+        provided direction, features, and language.
+
+        This is the amount by which following text should be offset.
+        Text bounding box may extend past the length in some fonts,
+        e.g. when using italics or accents.
+
+        The result is returned as a float; it is a whole number if using basic layout.
+
+        :param text: Text to measure.
+        :param mode: Used by some graphics drivers to indicate what mode the
+                     driver prefers; if empty, the renderer may return either
+                     mode. Note that the mode is always a string, to simplify
+                     C-level implementations.
+
+        :param direction: Direction of the text. It can be 'rtl' (right to
+                          left), 'ltr' (left to right) or 'ttb' (top to bottom).
+                          Requires libraqm.
+
+        :param features: A list of OpenType font features to be used during text
+                         layout. This is usually used to turn on optional
+                         font features that are not enabled by default,
+                         for example 'dlig' or 'ss01', but can be also
+                         used to turn off default font features for
+                         example '-liga' to disable ligatures or '-kern'
+                         to disable kerning.  To get all supported
+                         features, see
+                         https://docs.microsoft.com/en-us/typography/opentype/spec/featurelist
+                         Requires libraqm.
+
+        :param language: Language of the text. Different languages may use
+                         different glyph shapes or ligatures. This parameter tells
+                         the font which language the text is in, and to apply the
+                         correct substitutions as appropriate, if available.
+                         It should be a `BCP 47 language code
+                         <https://www.w3.org/International/articles/language-tags/>`
+                         Requires libraqm.
+
+        :return: Width for horizontal, height for vertical text.
+        """
+        return (
+            self.font.getlength(text, mode == "1", direction, features, language) / 64
+        )
+
+    def getbbox(
```

Comment:
Would it be pedantic to say this should be called `getcbox`, since this uses FreeType's `FT_Glyph_Get_CBox`, not `FT_Outline_Get_BBox`, if some future version of Pillow would ever add a separate version for that?

Reply:
The idea is to return the bounding box of the drawn text, i.e. the rectangle of pixels that may be modified by `draw.text`, not just the union of glyph `CBox`es. In fact, this function doesn't even return the tightest possible bound on the glyph `CBox`es, since it also includes the pen line (i.e. the origin and glyph advance points).

I think the reason for calling it `getbbox` is that this is what it is most often called in Pillow code (see https://pillow.readthedocs.io/en/latest/search.html?q=bbox). Another possible name would be `getbounds` to avoid confusion with the similarly named concept in FreeType? However, this doesn't seem to be used anywhere else in code yet (see https://pillow.readthedocs.io/en/latest/search.html?q=bound).